core/src/main/java/com/linecorp/armeria/common/AbstractHttpMessageBuilder.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Locale;
+
+import org.reactivestreams.Publisher;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.internal.common.JacksonUtil;
+
+abstract class AbstractHttpMessageBuilder {
+
+    @Nullable
+    private HttpData content;
+
+    @Nullable
+    private Publisher<? extends HttpData> publisher;
+
+    @Nullable
+    protected HttpData getContent() {
+        return content;
+    }
+
+    @Nullable
+    protected Publisher<? extends HttpData> getPublisher() {
+        return publisher;
+    }
+
+    /**
+     * Sets the content for this response.
+     */
+    public AbstractHttpMessageBuilder content(HttpData content) {
+        requireNonNull(content, "content");
+        this.content = content;
+        return this;
+    }
+
+    /**
+     * Sets the {@link Publisher} for this response.
+     */
+    public AbstractHttpMessageBuilder content(Publisher<? extends HttpData> content) {
+        requireNonNull(content, "publisher");
+        checkState(this.content == null, "content has been set already");
+        publisher = content;
+        return this;
+    }
```

Comment:
As `HttpData` is binary data, we can't guess the `MediaType`. could we remove these two builder methods? I guess that an implicit `MediaType` can be confusing to users.

Reply:
ok, remove them.